.github/workflows/ci.yaml

```diff
@@ -1,66 +1,128 @@
+name: Continuous Integration
+
 on:
   push:
     branches: [main]
   pull_request:
     branches: [main]
   merge_group:
     types: [checks_requested]
-
-name: Continuous integration
+  paths-ignore:
+    - "**.md"
+    - "**.yml"
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  test:
-    name: Test Suite
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@stable
-      - run: cargo build --all --all-features
-      - run: cargo test --all-features -- --nocapture
-
   fmt:
-    name: Rustfmt
+    name: Code Formatting
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@nightly
+      - name: Checkout PR
+        uses: actions/checkout@v4
+
+      - name: Use Rust nightly with rustfmt
+        uses: dtolnay/rust-toolchain@nightly
         with:
           components: rustfmt
+
       - run: cargo fmt -- --config group_imports=StdExternalCrate,imports_granularity=Module --check
 
   clippy:
     name: Clippy
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@stable
+      - name: Checkout PR
+        uses: actions/checkout@v4
+
+      - name: Use Rust nightly with clippy
+        uses: dtolnay/rust-toolchain@stable
         with:
           components: clippy
+
       - run: cargo clippy --no-deps --all-features --all-targets -- -D warnings -D clippy::expect_used -D clippy::unwrap_used -D clippy::unimplemented
 
-  coverage:
+  build-test-artifacts:
+    name: Test Artifacts
+    runs-on: ubuntu-latest
```

Comment:
i would add `needs: [fmt, clippy ]`  here unless you want this build to happen regardless if those steps pass. 
usually, if they fail it means there's a problem with the PR changes that need to be addressed anyways

Reply:
I'll address this in another PR. Thanks!